dvc/config_schema.py

```diff
@@ -225,13 +226,9 @@ class RelPath(str):
         )
     },
     "state": {
-        "dir": str,
```

Comment:
Can bet my money that some users already are using it in their git history, will will result in ConfigErrors for those revisions. Let's keep these options for now, but make them defunct. In the future these will be useful for sharing state/index.